adoc/extensions/sycl_khr_queue_empty_query.adoc

```diff
@@ -0,0 +1,86 @@
+[[sec:khr-queue-empty-query]]
+= sycl_khr_queue_empty_query
+
+This extension allows developers to query the queue's emptiness, meaning if all
+commands submitted to a queue have been completed.
+
+[[sec:khr-queue-empty-query-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+[[sec:khr-queue-empty-query-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_QUEUE_EMPTY_QUERY# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+[[sec:khr-queue-empty-query-funct]]
+== New Queue Function to Query Emptiness
+
+This extension adds the following function to the [code]#sycl::queue# class,
+which provides information about the emptiness of the queue.
+
+'''
+
+.[apidef]#queue::khr_empty#
+[source,role=synopsis,id=api:queue-khr-empty]
+----
+bool khr_empty() const
+----
+
+_Returns:_ [code]#true# if all <<command,commands>> enqueued on this particular
+queue have completed, [code]#false# otherwize.
+
+{note} Since the implementation excute commands asynchronously, the returned
+value is a snapshot in time.
+{endnote}
+
+'''
+
+[[sec:khr-queue-empty-query-example]]
+== Example
+
+The example below demonstrates the usage of this extension.
+
+[source,,linenums]
+----
+#include <assert.h>
+#include <atomic>
+#include <iostream>
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue q;
+  assert(q.khr_empty());
+  std::atomic_bool start = false;
+
+  auto e1 = q.submit([&](sycl::handler &cgh) {
+    cgh.host_task([&]() {
+      // To avoid the loop being optimized away
+      int iter = 0;
+      while (start != true) {
+        iter++;
+      }
+      std::cout << iter << std::endl;
+    });
+  });
+  assert(!q.khr_empty());
+  auto e2 = q.single_task(e1, [=] {});
+  assert(!q.khr_empty());
+  start = true;
+  e2.wait();
+  assert(q.khr_empty());
```

Comment:
I think this is not correct.  Because the queue is out-of-order, it's possible that `e2` completes before `e1`.  Therefore, the first host task might not be complete at this time.

More generally ... this code reads more like a CTS test to me than an example.  Can you think of some short example that demonstrates a useful usage of `khr_empty`?

Reply:
>I think this is not correct. Because the queue is out-of-order, it's possible that e2 completes before e1. Therefore, the first host task might not be complete at this time.

e2 depend on e1. 

>More generally ... this code reads more like a CTS test to me than an example. Can you think of some short example that demonstrates a useful usage of khr_empty?

Good point. Let me cook something

Reply:
Modified the example. I think it's more in line with how people use the API in Cuda 
```
  // Pool of queues
  std::vector<sycl::queue> Qs;
  size_t n_queues = 12;
  for (size_t i = 0; i < n_queues; i++)
    Qs.push_back(sycl::queue(sycl::property_list{sycl::property::queue::in_order{}}));

  // Useful receipt for load-balancing
  auto it = std::find_if(Qs.begin(), Qs.end(),
                         [](const sycl::queue &q) { return q.khr_empty(); }
  );
  if (it != Qs.end()) {
    std::cout << "Empty queue Present!" << std::endl;
  } else {
    std::cout << "No empty queue." << std::endl;
  }
```

Reply:
> e2 depend on e1

Oh, I missed that.  Sorry.

> Good point. Let me cook something

Much better.  Thanks!